zomes/rea_plan/zome_idx_plan/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+/**
+ * Plan query indexes for plan DNA
+ *
+ * @package Holo-REA
+ * @since   2021-09-06
+ */
+use hdk_semantic_indexes_zome_derive::index_zome;
+use hc_zome_rea_plan_rpc::*;
+
+// :TODO: remove this; should not be necessary since all these types are imported
+// along with their entry_def! in dependent crates
+#[hdk_extern]
+fn entry_defs(_: ()) -> ExternResult<EntryDefsCallbackResult> {
+    Ok(EntryDefsCallbackResult::from(vec![
+        PathEntry::entry_def(),
+        PlanAddress::entry_def(),
+        ProcessAddress::entry_def(),
+        CommitmentAddress::entry_def(),
+    ]))
+}
+
+#[index_zome]
+struct Plan {
+    processes: Remote<process, planned_within>,
+    independent_demands: Remote<commitment, planned_within>,
+    // NOTE: not sure how to handle the fact that there are two kinds of relationships between Commitment and Plan
```

Comment:
@pospi multi relationships between two types hasn't yet been covered in the existing code, but it will come up again as we implement some more vf types.

Reply:
we've resolved this, and made better sense of it 👍